packages/features/timezone-buddy/components/TimeDial.tsx

```diff
@@ -0,0 +1,126 @@
+import { useContext } from "react";
+import { useStore } from "zustand";
+
+import type { Dayjs } from "@calcom/dayjs";
+import dayjs from "@calcom/dayjs";
+import { classNames } from "@calcom/lib";
+import type { DateRange } from "@calcom/lib/date-ranges";
+
+import { DAY_CELL_WIDTH } from "../constants";
+import { TBContext } from "../store";
+
+interface TimeDialProps {
+  timezone: string;
+  dateRanges?: DateRange[];
+}
+
+function isMidnight(h: number) {
+  return h <= 5 || h >= 22;
+}
+
+function isCurrentHourInRange({
+  dateRanges,
+  cellDate,
+  offset,
+}: {
+  dateRanges?: DateRange[];
+  cellDate: Dayjs;
+  offset: number;
+}) {
+  if (!dateRanges) return false;
+  const currentHour = cellDate.hour();
+
+  return dateRanges.some((time) => {
+    if (!time) null;
+
+    const startHour = dayjs(time.start).subtract(offset, "hour");
+    const endHour = dayjs(time.end).subtract(offset, "hour");
+
+    // If not same day number then we don't care
+
+    if (startHour.day() !== cellDate.day()) {
+      return false;
+    }
+
+    // this is a weird way of doing this
+    const newDate = dayjs(time.start).set("hour", currentHour);
+
+    return newDate.isBetween(startHour, endHour, undefined, "[)"); // smiley faces or something
+  });
+}
+
+export function TimeDial({ timezone, dateRanges }: TimeDialProps) {
+  const store = useContext(TBContext);
+  if (!store) throw new Error("Missing TBContext.Provider in the tree");
+  const browsingDate = useStore(store, (s) => s.browsingDate);
+
+  const usersTimezoneDate = dayjs(browsingDate).tz(timezone);
+
+  const nowDate = dayjs(browsingDate);
+
+  const offset = (nowDate.utcOffset() - usersTimezoneDate.utcOffset()) / 60;
+
+  const hours = Array.from({ length: 24 }, (_, i) => i - offset + 1);
```

Comment:
Ensure the hours are correct basedo n the UTC offset diff from the nowDate of the provider and the timezone of the current row users tz